premerge/premerge-advisor.md

```diff
@@ -0,0 +1,168 @@
+# Premerge Advisor
+
+## Introduction
+
+While the premerge infrastructure is generally reliable, tests that are broken
+at HEAD, or tests that are flaky can significantly degrade the user experience.
+Failures unrelated to the PR being worked on can lead to warning fatigue which
+can compound this problem when people land failing tests into main that they
+believe are unrelated. The premerge advisor is designed to run after premerge
```

Comment:
I would suggest inserting a sentence: 'To address these issues we propose to implement a "premerge advisor".'  I'd put that just before the sentence that starts "The premerge advisor is..